docs/reference/query-languages/esql/esql-lookup-join.md

```diff
@@ -26,25 +28,31 @@ For example, you can use `LOOKUP JOIN` to:
 * You want to restrict users to use only specific lookup indices
 * You do not need to match using ranges or spatial relations
 
-## How the `LOOKUP JOIN` command works [esql-how-lookup-join-works]
+## How the command works [esql-how-lookup-join-works]
+
+The `LOOKUP JOIN` command adds fields from the lookup index as new columns to your results table based on matching values in the join field.
+
+The command requires two parameters:
+- The name of the lookup index (which must have the `lookup` [`index.mode setting`](/reference/elasticsearch/index-settings/index-modules.md#index-mode-setting))
+- The name of the field to join on
 
-The `LOOKUP JOIN` command adds new columns to a table, with data from {{es}} indices.
+```esql
+LOOKUP JOIN <lookup_index> ON <field_name>
+```
 
 :::{image} ../images/esql-lookup-join.png
-:alt: esql lookup join
+:alt: Illustration of the `LOOKUP JOIN` command, where the input table is joined with a lookup index to create an enriched output table.
 :::
 
-`<lookup_index>`
-: The name of the lookup index. This must be a specific index name - wildcards, aliases, and remote cluster references are not supported. Indices used for lookups must be configured with the [`lookup` index mode](/reference/elasticsearch/index-settings/index-modules.md#index-mode-setting).
+## Example
 
-`<field_name>`
-: The field to join on. This field must exist in both your current query results and in the lookup index. If the field contains multi-valued entries, those entries will not match anything (the added fields will contain `null` for those rows).
+If you're familiar with SQL, `LOOKUP JOIN` has left-join behavior. This means that if no rows match in the lookup index, the incoming row is retained and `null`s are added. If many rows in the lookup index match, `LOOKUP JOIN` adds one row per match.
 
-## Example
+### Sample tables
 
-`LOOKUP JOIN` has left-join behavior. If no rows match in the lookup index, `LOOKUP JOIN` retains the incoming row and adds `null`s. If many rows in the lookup index match, `LOOKUP JOIN` adds one row per match.
+In this example, we have two sample tables, `employees` and `languages_non_unique_key`. The `employees` table contains employee information, including their language code. The `languages_non_unique_key` table contains language codes and their corresponding names and countries.
```

Comment:
Employees don't have a language code, we construct this using a modulus operation just to have some way to correlate the two tables. If we want to stick with this example, we can maybe reword like this:
```suggestion
In this example, we have two sample tables, `employees` and `languages_non_unique_key`.  The `languages_non_unique_key` table contains languages together with countries where the language is spoken and a code that uniquely identifies the language. The `employees` table contains employee information, including their employee numbers.
```

And below, under `Query`, we could write:
```
For the sake of the example, we construct some arbitrary code from employee numbers and use it to assign a language based on this code.
```

Reply:
Thanks @alex-spies, I'm thinking for the sake of this illustrative example we can just rename the table to `languages`?

Reply:
Oh, right! I thought this example was included via `:::{include} ../examples/docs-lookup-join.csv-spec...` but it's actually hand-written.

I guess we can rename the table for now; I wouldn't polish this too much, though, given that we want to use a nicer example, anyway.

(And, ideally, that example should live in our tests and be included rather than being hand-written again. I'm happy to help place the example into our tests once we have it :) )

Reply:
blown away by https://github.com/elastic/elasticsearch/pull/127215/commits/9d23f244c7e200ff279ff5f2d245c9d5a53db50c